File: app/src/main/java/com/aire/android/lifecycle/LifecycleActivity.java
package com.aire.android.lifecycle;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.aire.android.test.R;

public class LifecycleActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_lifecycle);
        getLifecycle().addObserver(new LifecyclePresenter());
    }
}

File: app/src/main/java/com/aire/android/main/LifeCycleActivity.kt
package com.aire.android.main

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.aire.android.test.R

class LifeCycleActivity : Activity() {
    companion object {
        const val TAG = "lifecycle_activity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_life_cycle)
    }

    override fun onStart() {
        super.onStart()
        Log.i(TAG, "onStart: ")
    }

    override fun onResume() {
        super.onResume()
        Log.i(TAG, "onResume: ")
    }

    override fun onPause() {
        super.onPause()
        Log.i(TAG, "onPause: ")
    }

    override fun onStop() {
        super.onStop()
        Log.i(TAG, "onStop: ")
    }
}